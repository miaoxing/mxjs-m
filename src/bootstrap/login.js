module.exports = require('./login.' + (process.env.TARO_ENV || 'weapp'));
