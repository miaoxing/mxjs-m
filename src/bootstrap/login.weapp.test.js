import '@mxjs/taro';
import login from './login.weapp';
import Taro from '@tarojs/taro';
import $, {Ret} from 'miaoxing';
import {createPromise} from '@mxjs/test';
import {waitFor} from '@testing-library/react';

describe('login weapp', () => {
  test('suc', async () => {
    Taro.login = jest.fn().mockResolvedValueOnce({
      code: 'test-code',
    });

    $.http = jest.fn().mockResolvedValueOnce({
      ret: Ret.suc({token: 'test-token'}),
    });

    Taro.setStorageSync = jest.fn();

    await login();
    await waitFor(() => {
      expect(Taro.setStorageSync).toBeCalled();
    });

    expect(Taro.login.mock.calls).toMatchSnapshot();
    expect($.http.mock.calls).toMatchSnapshot();
    expect(Taro.setStorageSync.mock.calls).toMatchSnapshot();
  });

  test('Taro.login fail', async () => {
    const promise = createPromise();
    Taro.login = jest.fn().mockImplementationOnce(() => promise.resolve({
      errMsg: '测试登录失败',
    }));

    $.alert = jest.fn();

    await login();
    await promise;

    expect(Taro.login.mock.calls).toMatchSnapshot();
    expect($.alert.mock.calls).toMatchSnapshot();
  });

  test('HTTP fail', async () => {
    Taro.login = jest.fn().mockResolvedValueOnce({
      code: 'test-code',
    });

    const promise = createPromise();
    $.http = jest.fn().mockImplementationOnce(() => promise.resolve({
      ret: Ret.err('HTTP error'),
    }));

    $.alert = jest.fn();

    await login();
    await promise;

    expect(Taro.login.mock.calls).toMatchSnapshot();
    expect($.http.mock.calls).toMatchSnapshot();
    expect($.alert.mock.calls).toMatchSnapshot();
  });
});
