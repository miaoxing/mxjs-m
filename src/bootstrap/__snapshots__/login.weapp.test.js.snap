// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login weapp HTTP fail 1`] = `
Array [
  Array [],
]
`;

exports[`login weapp HTTP fail 2`] = `
Array [
  Array [
    Object {
      "data": Object {
        "code": "test-code",
      },
      "method": "post",
      "skipBeforeEvent": true,
      "url": "/m-api/wechat-mp/login",
    },
  ],
]
`;

exports[`login weapp HTTP fail 3`] = `
Array [
  Array [
    "HTTP error",
  ],
]
`;

exports[`login weapp Taro.login fail 1`] = `
Array [
  Array [],
]
`;

exports[`login weapp Taro.login fail 2`] = `
Array [
  Array [
    "微信登录失败：测试登录失败",
  ],
]
`;

exports[`login weapp suc 1`] = `
Array [
  Array [],
]
`;

exports[`login weapp suc 2`] = `
Array [
  Array [
    Object {
      "data": Object {
        "code": "test-code",
      },
      "method": "post",
      "skipBeforeEvent": true,
      "url": "/m-api/wechat-mp/login",
    },
  ],
]
`;

exports[`login weapp suc 3`] = `
Array [
  Array [
    "token",
    "test-token",
  ],
]
`;
